---
layout: ../../layouts/NewsLayout.astro
title: "Pathoplexus November Update"
---

<div>
    <a href="/news">News</a> > <b>Pathoplexus November Update</b>
</div>

# Pathoplexus November Update

*By the Pathoplexus Team - 18 November 2025*

Since our last update in August, Pathoplexus has continued to grow - both in data and in community. Over the past two months, thousands of new sequences have been shared, several new features have gone live, and new members have joined our global community! In addition, we were excited to be part of the joint PHA4GE & IPSN Conference - read on to find out more!


## Data and Pathogen Updates

Pathoplexus now hosts over 8,900 directly-submitted sequences, including new uploads of:

- 1,366 new mpox sequences, from [Pathogen Genomic Lab, DRC](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1756311091&isRevocation=&visibility_groupName=true&visibility_groupId=true&groupId=232&releasedAtTimestampTo=1762559999); [INSA, Portugal](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1756717628&releasedAtTimestampTo=1762559999&isRevocation=&visibility_groupName=true&visibility_groupId=true&groupId=265); [National Virus Reference Laboratory, Ireland](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1757693423&releasedAtTimestampTo=1762559999&isRevocation=false&visibility_groupId=true&groupId=463); [Laboratoire de Biologie moléculaire et d'Immunologie, Togo](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1758499200&releasedAtTimestampTo=1762559999&isRevocation=&visibility_groupId=true&groupId=530); [Public Health Agency of Canada, Canada](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1756944000&releasedAtTimestampTo=1762559999&isRevocation=&visibility_groupId=true&groupId=397); [Institute Pasteur de Dakar, Senegal](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1759862336&isRevocation=false&visibility_groupId=true&visibility_groupName=true&releasedAtTimestampTo=1762559999&geoLocCountry=Senegal&groupId=562); National Public Health Agency & Sierra Leone Ministry of Health, [with Broad and Scripps Institute](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupName=true&visibility_groupId=true&visibility_accessionVersion=true&groupId=498&visibility_earliestReleaseDate=true&visibility_version=true&visibility_submittedAtTimestamp=true&visibility_versionStatus=true&earliestReleaseDateFrom=2025-09-17&earliestReleaseDateTo=2025-11-07) and [with Institute Pasteur, Senegal and CDC](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1759276800&releasedAtTimestampTo=1762559999&isRevocation=&geoLocCountry=Sierra+Leone&visibility_groupId=true&groupId=562), Sierra Leone; [Institute National de Recherche Biomédicale, DRC](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&groupId=232&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-10-07&earliestReleaseDateTo=2025-11-07); and [National Genomic Sequencing Reference Laboratory, Malawi](https://pathoplexus.org/mpox/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupName=true&visibility_groupId=true&groupId=595&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-10&earliestReleaseDateTo=2025-11-07)

- 7 new West-Nile sequences, from [Infectious & Tropical Diseases Unit, Annunziata Hospital, Italy](https://pathoplexus.org/west-nile/search?geoLocCountry=Italy&visibility_releasedAtTimestamp=true&releasedAtTimestampFrom=1756339200&releasedAtTimestampTo=1757289599&visibility_groupId=true&groupId=431) and [Unité des Virus Émergents, Aix-Marseille Université, France](https://pathoplexus.org/west-nile/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupName=true&visibility_groupId=true&groupId=497&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-01&earliestReleaseDateTo=2025-11-07)

- 68 new Ebola-Zaire sequences, including those from past samples and from the recent outbreak, from [Institut National de Recherche Biomédicale, DRC](https://pathoplexus.org/ebola-zaire/search?visibility_releasedAtTimestamp=true&isRevocation=false&visibility_groupId=true&groupId=232&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-10&earliestReleaseDateTo=2025-11-07)

- 1,326 new RSV-A sequences, from [Grubaugh Lab, Yale, USA](https://pathoplexus.org/rsv-a/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&groupId=554&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-10&earliestReleaseDateTo=2025-11-07); [Boston University & Boston Medical Center, Massachusetts, USA](https://pathoplexus.org/rsv-a/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&groupId=556&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-02&earliestReleaseDateTo=2025-11-07); [WHO Influenza Collaborating Center, Australia (including samples from the Philippines)](https://pathoplexus.org/rsv-a/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&groupId=539&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-02&earliestReleaseDateTo=2025-11-07); and [Computational Evolution Group, ETH Zurich, Switzerland](https://pathoplexus.org/rsv-a/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-10&earliestReleaseDateTo=2025-11-07&visibility_groupName=true&groupName=Computational+Evolution+Group%2C+ETH+Zurich)

- 1,642 new RSV-B sequences, from [Grubaugh Lab, Yale, USA](https://pathoplexus.org/rsv-b/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&groupId=554&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-01&earliestReleaseDateTo=2025-11-07); [Boston University & Boston Medical Center, Massachusetts, USA](https://pathoplexus.org/rsv-b/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&groupId=556&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-01&earliestReleaseDateTo=2025-11-07); and [WHO Influenza Collaborating Center, Australia (including samples from the Philippines)](https://pathoplexus.org/rsv-b/search?visibility_releasedAtTimestamp=true&isRevocation=&visibility_groupId=true&groupId=539&visibility_earliestReleaseDate=true&earliestReleaseDateFrom=2025-09-01&earliestReleaseDateTo=2025-11-07)

As well as directly uploaded sequences, we also ingest data from INSDC databases, ensuring users can access all openly available sequences via Pathoplexus. 

#### A note on RSV

We have added 2,325 RSV-A and 1,335 RSV-B additional sequences from INSDC that are misclassified in NCBI taxonomy (under bovine orthopneumovirus) and were, hence, previously missed by our pipeline.

<details>

<summary>Click for more details on the RSV ingest update</summary>

To help ensure we could capture all available RSV sequences, we recently updated our RSV ‘minimizer’ (used to classify sequences).
Previously we were ingesting from taxon ID 11250, which is annotated as "human respiratory syncytial virus".
We noticed that some human sequences have been annotated with taxon ID 12814 ("respiratory syncytial virus"), which is currently listed in NCBI Taxonomy as a child of "Bovine orthopneumovirus" (taxon ID 11246) and not under 11250.
By updating the classifier we were able to ingest from the full set of sequences under Orthopneumovirus ([taxon ID 1868215](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?command=show&mode=node&id=1868215&lvl=3)), whilst filtering out bovine sequences based on sequence similarity.
We have let NCBI know about this misclassification.

</details>


## Community Highlights

Pathoplexus is dedicated to serving the pathogen community, and as such we aim to be active participants in initiatives that help us connect with and better understand the needs of our users (and potential users!). Thus, we’re exceptionally pleased to announce that we are now an [**institutional member of the International Pathogen Surveillance Network (IPSN)**](https://www.who.int/initiatives/international-pathogen-surveillance-network/membership), convened by the WHO. We are excited to have the opportunity to connect with pathogen and public health researchers and workers worldwide and strengthen our ties with global partners in pathogen data infrastructure. 

In addition, we’re grateful to the active pathogen communities for their work in improving standards and datasets. Recently, the mpox and RSV communities have been particularly active:

- [Newly designated mpox lineages](https://github.com/nextstrain/nextclade_data/tree/2025-09-09--12-13-13Z/data/nextstrain/mpox) are now annotated on Pathoplexus following community review 

- [New RSV-B lineages](https://github.com/nextstrain/nextclade_data/tree/2025-09-09--12-13-13Z/data/nextstrain/rsv) have also been integrated, improving consistency and downstream analyses across datasets. 

#### Citations

Additionally, we’d like to remind all those who use data from Pathoplexus that we do require [use of Pathoplexus SeqSets and DOIs](https://pathoplexus.org/docs/how-to/generate-seqset) for citations in publications and preprints (this is especially critical when making use of [Restricted-Use Data](https://pathoplexus.org/about/terms-of-use/data-use-terms#42-restricted-data-use) in any form). These DOIs should be cited as references to enable tracking; a citation-style reference can be generated using the ‘Export’ button at the top of the page. 

Not only does proper citation ensure submitters receive full credit, but it also helps trace data provenance and use through time. Soon, this will allow submitters to see the total contribution their sequences have made towards work outside their own publications!

![Image showing the location of the links to generate and cite a DOI on the SeqSet page](/gifs/news/2025-11-27-november-update/seqset_combined.png)

## Technical Improvements

A series of backend and frontend enhancements have rolled out this quarter to make Pathoplexus faster, clearer, and more user-friendly:

### Submission:

- Accented characters are [now supported](https://github.com/loculus-project/loculus/commit/ea932f3c2065eb59bbabb0141577cd99f1d42d5c) in author names. Note that the INSDC accepts only ASCII characters. When submitting sequences to ENA, we hence replace non-ASCII characters with their nearest ASCII equivalent.

- For users uploading measles sequences, we have a new field `means2id` that can be used to link sequences to the [MeaNS2 database](https://who-gmrln.org/means2), if the sequences are also uploaded there, aiding with de-duplication.

- When submitting data to Pathoplexus as ‘Open,’ we now have a very clear pop-up to prevent submission without accidentally forgetting to change the Data Use Terms
![Image showing the pop-up that now appears when you submit data as Open, reminding users that once they submit data as 'Open' it cannot be changed to Restricted-Use](/gifs/news/2025-11-27-november-update/openData_release_alert.png)

### Display:

- Improved banners on Restricted-Use Data sequence pages now provide more detail about usage terms 
![Image showing the new Restricted-Use banner (in yellow) that appears on every Restricted-Use sequence page](/gifs/news/2025-11-27-november-update/restricted-use-banner.png)

- Group pages now show uploaded sequences (for logged-in users), giving users a clearer view of the data that’s been shared by a particular group - check it out on the [Institut National de Recherche Biomédicale Group Page](https://pathoplexus.org/group/232) *(you’ll need to log-in first!)*
![Image showing the total number of sequences for each pathogen uploaded by a group. For this group there are 68 Ebola Zaire and 2002 Mpox sequences](/gifs/news/2025-11-27-november-update/grouppageDetail.jpg)

### Access & Analysis:

- FASTA downloads with rich headers are now 20× faster and can be compressed, reducing transfer time for large datasets.

- Many search fields now support multiple-field search - allowing you to look at multiple countries, groups, and more at once!
![Image showing the search bar in Pathoplexus, with both 'Pathogen Genomic Lab' and 'KGHSL_VHF_LAB' searched as submitting group, and 'Democratic Republic of the Congo' and 'Sierra Leone' searched as the country](/gifs/news/2025-11-27-november-update/searchFields-multi-select.png)

- You can now view a list of external tools that make use of Pathoplexus data on each organism's 'Tools' page. Interested in listing your own tool? We welcome integrating additional tools! Please [open an issue](https://github.com/pathoplexus/pathoplexus/issues) to let us know! *(Tools can also be for just one or a selection of pathogens)*
![Image showing the Tools page for RSV-A, which lists GenSpectrum, Nextstrain, and Pathogen Sequence Counts and Maps as resources making use of Pathoplexus data for RSV-A via the API](/gifs/news/2025-11-27-november-update/externalTools.jpg)


## Talks, Tutorials, and Media

#### Past talks

Pathoplexus and Loculus have been on the road - or rather, on the conference circuit!

- The [Binfie Podcast](https://microbinfie.github.io/) released a two-part feature on Pathoplexus and Loculus - covering their design philosophy, technical underpinnings, and vision for open pathogen data. Have a listen for yourself: [Part I](https://soundcloud.com/microbinfie/micro-binfie-pathoplexus-part/s-rnGclEiarcS), [Part II](https://soundcloud.com/microbinfie/144-micro-binfie-pathoplexus-part-2/s-IvdaQga0ibJ) 

- The Loculus tutorial and talk at [BC2](https://www.bc2.ch/) by Chaoran Chen and Anna Parker drew strong engagement, with many participants expressing interest in federated deployments.

- Theo Sanderson presented Pathoplexus at [IMMEM](https://www.escmid.org/congress-events/immem-xiv/), highlighting how open-source infrastructure and transparent governance can power equitable data sharing.

- Chaoran Chen presented Pathoplexus at the CERI Summer Scientific Seminar at Stellenbosch University.

- Pathoplexus featured heavily at the [PHA4GE-Conference](https://pha4ge.org/conferences/pha4ge-conference-2025/) pre-conference workshop and at the conferences itself! The well-attended workshop featured a diverse group getting to grips with how to create accounts, search Pathoplexus, and upload data - and was led by Arthur Shem Kasambula, Stephen Kanyerezi, Patricia Nabisubi (Uganda) and Emma Hodcroft and Chaoran Chen (Switzerland). The talk focused on exploring how open data ecosystems can serve public health. Here are a few photos of the coordinators and workshop in action!
![A collage image, showing first a photo of the workshop coordinators Arthur, Stephen, Patricia, Emma, and Chaoran next to a PHA4GE-Workshop poster. Then showing Stephen helping a participant, and then Chaoran helping a participant](/gifs/news/2025-11-27-november-update/workshop-photo.jpg)

- Cornelius Roemer presented Pathoplexus at the 2025 ENA facilities on November 7.

#### Upcoming talks

- Emma Hodcroft will talk about Pathoplexus at the [European Virus Bioinformatics Center "Viruses in Silico"](https://evbc.uni-jena.de/events/viruses-in-silico/) lecture series on 27 Nov at 16.00 CET

## Awards and Collaborations

We are incredibly pleased to announce that Loculus, the software powering Pathoplexus, was chosen as an [SIB Remarkable Output 2024](https://www.sib.swiss/news/discover-the-sib-remarkable-outputs-2024)! In particular, judges [commented](https://bsky.app/profile/sib.swiss/post/3lw4drsnknc2s), “Loculus is remarkable for enabling efficient and secure sharing of pathogen genomic data, which is crucial for public health responses to infectious diseases."

In addition, the Loculus team is excited to be joining the [Pathogen Data Network](https://pathogendatanetwork.org/) to aid more efficient and effective sharing of pathogen data. This initiative will help us expand and improve Loculus. Since Loculus is the software that underpins Pathoplexus - this means more features and benefits for Pathoplexus, as well!

## New members

We’re pleased to welcome 8 new members to Pathoplexus this month: Vitor Borges (Portugal), Nick Goldman (UK), Arthur Shem Kasambula (Uganda), Stephen Kanyerezi (Uganda), Alejandra Gonzalez Sanchez (Spain), Laura Fahey (Ireland), Alan Rice (Ireland), and Senjuti Saha (Bangladesh). 

### Meetings, Minutes, and Resolutions

- Remember that you can always check our [minutes from General Assemblies and Executive Board meetings, as well as Executive Board resolutions](https://pathoplexus.org/about/governance/minutes)
